Index: src/Views/Home/HomeView.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?php\n\nnamespace Views\\Home;\nuse Controllers\\User\\Login;\nuse Models\\User\\User;\nuse Views\\AbstractView;\nuse Views\\Base\\BaseView;\n\nclass HomeView extends BaseView {\n\n    public const USERNAME_KEY = 'USERNAME_KEY';\n    public const LINK_KEY = 'LINK_KEY';\n\n    private const TEMPLATE_HTML = __DIR__ . '/Home.html';\n\n    public function __construct(?User $user = null) {\n        parent::__construct();\n        $this->setUser($user);\n\n    }\n\n    public function templatePath() : string {\n        return self::TEMPLATE_HTML;\n    }\n\n    public function templateKeys() : array {\n\n        $isLogged = $this->user !== null;\n        return [\n            self::USERNAME_KEY => $this->user?->getUsername() ?? 'Se connecter',\n            self::LINK_KEY => $isLogged  ? '/user/logout' : Login::PATH,\n        ] ;\n    }\n}
===================================================================
diff --git a/src/Views/Home/HomeView.php b/src/Views/Home/HomeView.php
--- a/src/Views/Home/HomeView.php	(revision f2873ccdbe58ddbc55f18a972187139629b532e5)
+++ b/src/Views/Home/HomeView.php	(date 1758621160813)
@@ -2,7 +2,7 @@
 
 namespace Views\Home;
 use Controllers\User\Login;
-use Models\User\User;
+use Models\User\UserDTO;
 use Views\AbstractView;
 use Views\Base\BaseView;
 
@@ -13,7 +13,7 @@
 
     private const TEMPLATE_HTML = __DIR__ . '/Home.html';
 
-    public function __construct(?User $user = null) {
+    public function __construct(?UserDTO $user = null) {
         parent::__construct();
         $this->setUser($user);
 
@@ -23,12 +23,8 @@
         return self::TEMPLATE_HTML;
     }
 
-    public function templateKeys() : array {
-
-        $isLogged = $this->user !== null;
-        return [
-            self::USERNAME_KEY => $this->user?->getUsername() ?? 'Se connecter',
-            self::LINK_KEY => $isLogged  ? '/user/logout' : Login::PATH,
-        ] ;
+    public function templateKeys() : array
+    {
+        return [];
     }
 }
\ No newline at end of file
Index: src/Views/Base/BaseView.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?php\n\nnamespace Views\\Base;\n\nuse Models\\User\\User;\nuse Views\\AbstractView;\n\nabstract class BaseView extends AbstractView\n{\n\n    private HeaderView $header;\n    private FooterView $footer;\n\n    protected ?User $user;\n    public function __construct()\n    {\n       $this->user = null ;\n\n    }\n\n    function render(){\n        $this->header = new HeaderView($this->user);\n        $this->footer = new FooterView();\n        $this->header->renderBody();\n        $this->renderBody();\n        $this->footer->renderBody();\n    }\n\n    function setUser(?User $user){\n        $this->user = $user;\n    }\n}
===================================================================
diff --git a/src/Views/Base/BaseView.php b/src/Views/Base/BaseView.php
--- a/src/Views/Base/BaseView.php	(revision f2873ccdbe58ddbc55f18a972187139629b532e5)
+++ b/src/Views/Base/BaseView.php	(date 1758621160780)
@@ -2,7 +2,7 @@
 
 namespace Views\Base;
 
-use Models\User\User;
+use Models\User\UserDTO;
 use Views\AbstractView;
 
 abstract class BaseView extends AbstractView
@@ -11,7 +11,7 @@
     private HeaderView $header;
     private FooterView $footer;
 
-    protected ?User $user;
+    protected ?UserDTO $user;
     public function __construct()
     {
        $this->user = null ;
@@ -26,7 +26,7 @@
         $this->footer->renderBody();
     }
 
-    function setUser(?User $user){
+    function setUser(?UserDTO $user){
         $this->user = $user;
     }
 }
\ No newline at end of file
Index: src/Views/Base/HeaderView.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?php\n\nnamespace Views\\Base;\n\nuse Models\\User\\User;\nuse Views\\AbstractView;\n\nclass HeaderView extends AbstractView\n{\n\n    public function __construct(?User $user = null)\n    {\n    }\n\n    function templatePath(): string\n    {\n        return __DIR__ . '/header.html';\n    }\n\n    function templateKeys(): array\n    {\n        return [];\n    }\n}
===================================================================
diff --git a/src/Views/Base/HeaderView.php b/src/Views/Base/HeaderView.php
--- a/src/Views/Base/HeaderView.php	(revision f2873ccdbe58ddbc55f18a972187139629b532e5)
+++ b/src/Views/Base/HeaderView.php	(date 1758621160796)
@@ -2,13 +2,21 @@
 
 namespace Views\Base;
 
-use Models\User\User;
+use Controllers\User\Login;
+use Models\User\UserDTO;
 use Views\AbstractView;
 
+
+
 class HeaderView extends AbstractView
 {
 
-    public function __construct(?User $user = null)
+    public const LINK_KEY = 'LINK_KEY';
+
+    public const USERNAME_KEY = 'USERNAME_KEY';
+
+
+    public function __construct(?UserDTO $user = null)
     {
     }
 
@@ -19,6 +27,10 @@
 
     function templateKeys(): array
     {
-        return [];
+        $isLogged = $this->user !== null;
+        return [
+            self::USERNAME_KEY => $this->user?->getUsername() ?? 'Nom Prenom',
+            self::LINK_KEY => $isLogged  ? '/user/logout' : Login::PATH,
+        ] ;
     }
 }
\ No newline at end of file
Index: src/Views/Base/header.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><!DOCTYPE html>\n<html lang=\"en\">\n<head>\n    <meta charset=\"UTF-8\">\n    <meta http-equiv=\"X-UA-Compatible\" content=\"IE=edge\">\n    <meta name=\"viewport\" content=\"width=device-width, initial-scale=1.0\" >\n    <link rel=\"stylesheet\" href=\"/_assets/css/index.css\">\n    <title>Document</title>\n</head>\n<body>
===================================================================
diff --git a/src/Views/Base/header.html b/src/Views/Base/header.html
--- a/src/Views/Base/header.html	(revision f2873ccdbe58ddbc55f18a972187139629b532e5)
+++ b/src/Views/Base/header.html	(date 1758620911313)
@@ -5,6 +5,18 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1.0" >
     <link rel="stylesheet" href="/_assets/css/index.css">
-    <title>Document</title>
+    <title>SAE Manager</title>
 </head>
-<body>
\ No newline at end of file
+<body>
+<header class="header">
+    <div class="header-content">
+        <div class="logo">
+            <h1>SAE Tracker</h1>
+            <span>Département Informatique - AMU</span>
+        </div>
+        <div class="user-info">
+            <span>👤{{USERNAME_KEY}}</span>
+            <button class="btn btn-outline" onclick="logout()"><a href="{{LINK_KEY}}">Se connecter</a> </button>
+        </div>
+    </div>
+</header>
Index: src/Controllers/Home/HomeController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?php\nnamespace Controllers\\Home;\n\nuse Controllers\\ControllerInterface;\nuse Models\\User\\User;\nuse Views\\Home\\HomeView;\n\nclass HomeController implements ControllerInterface\n{\n    function control(){\n        $user= new User('Toto', 'pass') ;\n        $view = new HomeView($user);\n        $view->render();\n    }\n\n    static function support(string $chemin, string $method) : bool{\n        return $chemin === \"/\" && $method === \"GET\";\n    }\n}
===================================================================
diff --git a/src/Controllers/Home/HomeController.php b/src/Controllers/Home/HomeController.php
--- a/src/Controllers/Home/HomeController.php	(revision f2873ccdbe58ddbc55f18a972187139629b532e5)
+++ b/src/Controllers/Home/HomeController.php	(date 1758621160851)
@@ -2,13 +2,13 @@
 namespace Controllers\Home;
 
 use Controllers\ControllerInterface;
-use Models\User\User;
+use Models\User\UserDTO;
 use Views\Home\HomeView;
 
 class HomeController implements ControllerInterface
 {
     function control(){
-        $user= new User('Toto', 'pass') ;
+        $user= new UserDTO('Toto', 'pass') ;
         $view = new HomeView($user);
         $view->render();
     }
